core/Cargo.toml

```diff
@@ -43,6 +44,7 @@ pretty_dtoa = "0.3"
 mime = "0.3"
 reqwest = { version = "^0.11", optional = true }
 bytes = { version = "^1.3", optional = true }
+url = { version = "2.4.1", optional = true }
```

Comment:
The `iref` crate already handles URLs through the [`Uri`/`UriBuf` type](https://docs.rs/iref/latest/iref/uri/struct.Uri.html).